lib/solidus_kustomer/client.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module SolidusKustomer
+  class Client
+    class << self
+      def from_config
+        new(api_key: SolidusKustomer.configuration.api_key)
+      end
+    end
+
+    def initialize(api_key:, url: 'https://api.kustomerapp.com/v1/')
+      @api_key = api_key
+      @url = url
+    end
+
+    # Create a new instance of a Kustomer Klass for a customer
+    #
+    # @param kobject_klass [String] the Klass name of the KObject to create
+    # @param for [Spree::User] the user whose id will own the created KObject
+    #
+    # @return [Boolean] whether the creation succeded or not
+    def create(kobject_klass, attributes:, customer:)
+      customer_id = find_customerid_by_externalid(customer)
+
+      response = HTTParty.post(
+        "#{@url}customers/#{customer_id}/klasses/#{kobject_klass}",
+        body: {
+          title: kobject_klass.capitalize,
+          custom: attributes
+        }.to_json,
+        headers: headers
+      )
+
+      response.success? || raise(SolidusKustomer::CreateError)
+    end
+
+    # Finds the UUID of the customer via the Solidus database id
+    #
+    # @param user [Spree::User] the user whose Kustomer UUID we're looking for
+    #
+    # @return [String] the user Kustomer UUID
+    def find_customerid_by_externalid(user)
```

Comment:
I would add underscores before `id`: `find_customer_id_by_external_id`.

Reply:
The method name has been changed to `find_customer_by_external_id` to better reflect the new behaviour.